js/packages/wallet-adapter-mobile/src/adapter.ts

```diff
@@ -394,15 +394,11 @@ export class SolanaMobileWalletAdapter extends BaseMessageSignerWalletAdapter {
                         const [signedTransaction] = await wallet.signTransactions({
                             transactions: [transaction],
                         });
-                        if (isVersionedTransaction(signedTransaction)) {
-                            return await connection.sendTransaction(signedTransaction);
-                        } else {
-                            const serializedTransaction = signedTransaction.serialize();
-                            return await connection.sendRawTransaction(serializedTransaction, {
-                                ...options,
-                                preflightCommitment: getTargetCommitment(),
-                            });
-                        }
+                        const serializedTransaction = signedTransaction.serialize();
```

Comment:
Both `LegacyTransaction` and `VersionedTransaction` implement `serialize()` so there's no reason to disambiguate.